zUtilities/QuickSave.cpp

```diff
@@ -279,12 +279,12 @@ namespace GOTHIC_ENGINE {
   }
 
   void QuickSave::MenuLoop() {
-    if ( !Options::QuickSaveMode || zCMenu::inGameMenu || isLoading || !zinput->KeyToggled( Options::KeyQuickLoad ) ) return;
+    if ( !Options::QuickSaveMode || zCMenu::inGameMenu || isLoading ) return;
 
     oCSavegameInfo* info = gameMan->savegameManager->GetSavegame( iLastSaveSlot );
     if ( !info || !info->DoesSavegameExist() ) return;
 
-    LoadFromMainMenu();
+    if ( Options::LoadLastQuickSaveAfterGameStart || zinput->KeyToggled( Options::KeyQuickLoad ) ) LoadFromMainMenu();
```

Comment:
@Myszax It's probably a cleaner solution than mine (hooking to Game_Init with a delay), but wouldn't it also trigger when user opens menu during game by pressing Esc? How is it prevented? `zCMenu::inGameMenu` checks if user is in main menu, or `MenuLoop` runs only in main menu?

Reply:
`zCMenu::inGameMenu`:
is `true` when save has been loaded or new game has been started;
is `false` when gothic is running and no above action was executed

`MenuLoop` is running in main menu (before load/new game) and also when `ESC` is pressed when game is running (after load/new game).

In concluding this loop won't execute if save has been loaded/new game has been started.
Please read [#34](https://github.com/Franisz/zUtilities/pull/33#issuecomment-2304776242)
I hope it explains xD

Reply:
Then it seems fine, I don't see why it could be "problematic"

Reply:
In Gothic you cannot exit back to the menu once a save is loaded. How is that not problematic?

Reply:
I still don't understand the problem. Why would you need to go back to main menu? Autoload feature needs the user to be in main menu only once: after game is launched, but before save is loaded.

Reply:
Can you do something from main menu, that you can't from in-game menu? In-game menu is 1:1 the same as main menu. Even if it wasn't, there is always an option to disable the autoload feature.

Reply:
> In Gothic you cannot exit back to the menu once a save is loaded. How is that not problematic?

Why would you? :P When you enable in `gothic.ini` option to auto load after game start (which is disabled by default [in this pr]) what is the point to go back to main menu? You still have menu by pressing `ESC` in game when save is already loaded.